subsys/net/lib/http/http_server.c

```diff
@@ -0,0 +1,1235 @@
+/*
+ * Copyright (c) 2023, Emna Rekik
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "headers/server_functions.h"
+
+#include <errno.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <strings.h>
+
+#ifdef __linux__
+
+#define CONFIG_NET_HTTP_SERVER_MAX_RESPONSE_SIZE 128
+#define CONFIG_NET_HTTP_SERVER_MAX_URL_LENGTH    64
+#define CONFIG_NET_HTTP_SERVER_MAX_FRAME_SIZE    2048
+
+#endif
+
+#define HTTP_SERVER_MAX_RESPONSE_SIZE CONFIG_NET_HTTP_SERVER_MAX_RESPONSE_SIZE
+#define HTTP_SERVER_MAX_URL_LENGTH    CONFIG_NET_HTTP_SERVER_MAX_URL_LENGTH
+#define HTTP_SERVER_MAX_FRAME_SIZE    CONFIG_NET_HTTP_SERVER_MAX_FRAME_SIZE
+
+#if !defined(__ZEPHYR__) || defined(CONFIG_POSIX_API)
+
+#define INCLUDE_HTML_CONTENT 1
+
+#include <arpa/inet.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <poll.h>
+#include <sys/eventfd.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#else
+
+#include <zephyr/kernel.h>
+#include <zephyr/net/net_ip.h>
+#include <zephyr/net/socket.h>
+#include <zephyr/posix/sys/eventfd.h>
+
+#endif
+
+#if defined(__ZEPHYR__) || defined(CONFIG_POSIX_API)
+
+#define SERVER_IPV4_ADDR "192.0.2.1"
+#include <zephyr/data/json.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(net_http_server, LOG_LEVEL_DBG);
+#include <zephyr/net/tls_credentials.h>
+
+#elif defined(__linux__)
+
+#define SERVER_IPV4_ADDR "127.0.0.1"
+#include "headers/mlog.h"
+#include <jansson.h>
+
+#endif
+
+static char url_buffer[CONFIG_NET_HTTP_SERVER_MAX_URL_LENGTH];
+static char http_response[CONFIG_NET_HTTP_SERVER_MAX_RESPONSE_SIZE];
+static const char *preface = "PRI * HTTP/2.0\r\n\r\nSM\r\n\r\n";
+
+static struct http_parser_settings parserSettings;
+static struct http_parser parser;
+
+static unsigned char settings_frame[9] = {
+	0x00, 0x00, 0x00,        /* Length */
+	0x04,                    /* Type: 0x04 - setting frames for config or acknowledgment */
+	0x00,                    /* Flags: 0x00 - unused flags */
+	0x00, 0x00, 0x00, 0x00}; /* Reserved, Stream Identifier: 0x00 - overall connection */
+
+static unsigned char settings_ack[9] = {
+	0x00, 0x00, 0x00,        /* Length */
+	0x04,                    /* Type: 0x04 - setting frames for config or acknowledgment */
+	0x01,                    /* Flags: 0x01 - ACK */
+	0x00, 0x00, 0x00, 0x00}; /* Reserved, Stream Identifier */
+
+static const char content_200[] = {
+#ifdef INCLUDE_HTML_CONTENT
+#include "index.html.gz.inc"
+#endif
+};
+
+static const char content_404[] = {
+#ifdef INCLUDE_HTML_CONTENT
+#include "not_found_page.html.gz.inc"
+#endif
+};
+
+bool has_upgrade_header = true;
+
+#if defined(__ZEPHYR__)
+static const struct json_obj_descr arithmetic_result_descr[] = {
+	JSON_OBJ_DESCR_PRIM(struct arithmetic_result, x, JSON_TOK_NUMBER),
+	JSON_OBJ_DESCR_PRIM(struct arithmetic_result, y, JSON_TOK_NUMBER),
+	JSON_OBJ_DESCR_PRIM(struct arithmetic_result, result, JSON_TOK_NUMBER),
+};
+#endif
```

Comment:
This seems to be test-specific data in the generic http server code. I don't think it belongs here.